builtin_array.go

```diff
@@ -80,7 +80,19 @@ func relToIdx(rel, l int64) int64 {
 	return max(l+rel, 0)
 }
 
-func (r *Runtime) newArrayValues(values []Value) *Object {
+func (r *Runtime) newArrayValues(values valueStack) *Object {
+	if r.shouldForceMemCheck {
+		memCtx := newMemUsageContextClone()
+		if memCtx != nil {
+			memUsage, err := valuesMemUsage(values, memCtx)
+			if err != nil {
+				panic(err)
+			}
+			if memCtx.MemUsageLimitExceeded(memUsage) {
+				panic("memory limit exceeded")
```

Comment:
[q] would the panic know the func name when it happens? We can use the runtime's vm's func name if the stack trace doesn't help us here. I guess we would flag certain apps, so in theory we know which functions are causing the issue. This may help if there are multiple triggers and functions for flagged apps or if we have multiple apps flagged running at the same time.

Reply:
I need to see how I can bubble this up to baas, I still haven't ironed out how this comes through in baas so that we can _parse_ it and potentially log more details.

Reply:
yea, i agree reporting to baas could be a problem, e.g. how do we report if recursive array objects, where another array embedded into an array, do we only report max memUsage?

Reply:
Will delegate to the baas PR to figure this out. But in this case the panic will stop execution and for recursive data structures the poller eventually will catch it. Either way I'll have to test this out and report back. I'll leave this thread open until I have done more tests locally.